pkg/storage/driver/sql.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package driver
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	migrate "github.com/rubenv/sql-migrate"
+
+	// Import pq for potgres dialect
+	_ "github.com/lib/pq"
+
+	rspb "k8s.io/helm/pkg/proto/hapi/release"
+	storageerrors "k8s.io/helm/pkg/storage/errors"
+)
+
+var _ Driver = (*SQL)(nil)
+
+var labelMap = map[string]string{
+	"MODIFIED_AT": "modified_at",
+	"CREATED_AT":  "created_at",
+	"VERSION":     "version",
+	"STATUS":      "status",
+	"OWNER":       "owner",
+	"NAME":        "name",
+}
+
+// SQLDriverName is the string name of this driver.
+const SQLDriverName = "SQL"
+
+// SQL is the sql storage driver implementation.
+type SQL struct {
+	db  *sqlx.DB
+	Log func(string, ...interface{})
+}
+
+// Name returns the name of the driver.
+func (s *SQL) Name() string {
+	return SQLDriverName
+}
+
+func (s *SQL) ensureDBSetup() error {
+	// Populate the database with the relations we need if they don't exist yet
+	migrations := &migrate.MemoryMigrationSource{
+		Migrations: []*migrate.Migration{
+			{
+				Id: "init",
+				Up: []string{
+					`
+            CREATE TABLE releases (
+		      		key VARCHAR(67) PRIMARY KEY,
+              body TEXT NOT NULL,
+
+              name VARCHAR(64) NOT NULL,
+              version INTEGER NOT NULL,
+		      		status TEXT NOT NULL,
+		      		owner TEXT NOT NULL,
+		      		created_at INTEGER NOT NULL,
+		      		modified_at INTEGER NOT NULL DEFAULT 0
+            );
+
+            CREATE INDEX ON releases (key);
+            CREATE INDEX ON releases (version);
+            CREATE INDEX ON releases (status);
+            CREATE INDEX ON releases (owner);
+            CREATE INDEX ON releases (created_at);
+            CREATE INDEX ON releases (modified_at);
+		      `,
+				},
+				Down: []string{
+					`
+            DROP TABLE releases;
+          `,
+				},
+			},
+		},
+	}
+
+	_, err := migrate.Exec(s.db.DB, "postgres", migrations, migrate.Up)
+	return err
+}
+
+// Release describes a Helm release
+type Release struct {
```

Comment:
Is there some way we can use the Release object similar to the other storage drivers? I understand the need for the annotations for the sql driver to work, but this means we now have [two different ways of representing Helm releases](https://github.com/helm/helm/blob/33589472cf02d8d10e96e06085d95791c0611a75/pkg/proto/hapi/release/release.pb.go#L19) in the backend, which means that any time we introduce changes to the Release object, we'll also have to update this struct as well.

Reply:
I'm sorta inclined to go this way, since moving the other way would involve adding writing new packages that plug into the protobuf tools, which are already the most fragile piece of Helm. So I would be okay with this... and hopefully Helm 3 will make this much easier.

Reply:
Sounds good to me! Thanks for the clarification. :)